mkdocs/structure/toc.py

```diff
@@ -49,79 +54,9 @@ def indent_print(self, depth=0):
         return ret
 
 
-class _TOCParser(HTMLParser):
-    def __init__(self):
-        HTMLParser.__init__(self)
-        self.links = []
-
-        self.in_anchor = False
-        self.attrs = None
-        self.title = ''
-
-        # Prior to Python3.4 no convert_charrefs keyword existed.
-        # However, in Python3.5 the default was changed to True.
-        # We need the False behavior in all versions but can only
-        # set it if it exists.
-        if hasattr(self, 'convert_charrefs'):  # pragma: no cover
-            self.convert_charrefs = False
-
-    def handle_starttag(self, tag, attrs):
-        if not self.in_anchor:
-            if tag == 'a':
-                self.in_anchor = True
-                self.attrs = dict(attrs)
-
-    def handle_endtag(self, tag):
-        if tag == 'a':
-            self.in_anchor = False
-
-    def handle_data(self, data):
-        if self.in_anchor:
-            self.title += data
-
-    def handle_charref(self, ref):
-        self.handle_entityref("#" + ref)
-
-    def handle_entityref(self, ref):
-        self.handle_data("&%s;" % ref)
-
-
-def _parse_html_table_of_contents(html):
-    """
-    Given a table of contents string that has been automatically generated by
-    the markdown library, parse it into a tree of AnchorLink instances.
-
-    Returns a list of all the parent AnchorLink instances.
-    """
-    lines = html.splitlines()[2:-2]
-    ret, parents, level = [], [], 0
-    for line in lines:
-        parser = _TOCParser()
-        parser.feed(line)
-        if parser.title:
-            try:
-                href = parser.attrs['href']
-            except KeyError:
-                continue
-            title = parser.title
-            nav = AnchorLink(title, href, level)
-            # Add the item to its parent if required.  If it is a topmost
-            # item then instead append it to our return value.
-            if parents:
-                parents[-1].children.append(nav)
-            else:
-                ret.append(nav)
-            # If this item has children, store it as the current parent
-            if line.endswith('<ul>'):
-                level += 1
-                parents.append(nav)
-        elif line.startswith('</ul>'):
-            level -= 1
-            if parents:
-                parents.pop()
-
-    # For the table of contents, always mark the first element as active
-    if ret:
-        ret[0].active = True
-
-    return ret
+def _parse_toc_token(token):
+    # MkDocs uses 0-indexed levels, whereas the toc tokens are 1-indexed.
```

Comment:
> Currently, it's 0-indexed, but it might be better to make it 1-indexed so that the level matches N in <hN> for the headers.

I think I agree. While I'm used to working with 0-indexed objects, it might be confusing to non-programer theme devs because of the mismatch between the header level and the index level. Also, there is the possibility that a document could start with something other than an `h1`. If a template was testing for that, it could get really confusing. So let's change this.

Reply:
Ok, done. I'll leave this comment unresolved as a reminder to write a release note about this...

Reply:
Added a release note for this.